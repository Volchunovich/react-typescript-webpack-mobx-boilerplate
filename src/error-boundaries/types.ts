import { ErrorInfo } from 'react';

// region LogReportingService
export type LogReportingServiceArgsType = (error: Error, errorInfo: ErrorInfo) => void;
// endregion
