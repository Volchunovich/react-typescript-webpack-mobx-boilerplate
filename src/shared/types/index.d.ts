
// declare module '*.ttf' {
//   const a: string;
//
//   export = a;
// };

declare module '*.ttf';
