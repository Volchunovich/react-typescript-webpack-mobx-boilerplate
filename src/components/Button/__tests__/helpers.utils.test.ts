import { ButtonStyledProps } from '../utils/button.types';
import getTheme from '../utils/helpers.utils';

describe('Button', () => {
  describe('Helpers', () => {
    it("Should received 'primary' buttonType and return string", () => {
      const props: ButtonStyledProps = { buttonType: 'primary' };
      expect(props.buttonType).toBe('primary');
      expect(props.buttonType).toMatchSnapshot('primary');

      const tmp = getTheme(props);
      expect(typeof tmp).toEqual('string');
      expect(tmp).toMatchSnapshot();
    });

    it("Should received 'light' buttonType and return string", () => {
      const props: ButtonStyledProps = { buttonType: 'light' };
      expect(props.buttonType).toBe('light');
      expect(props.buttonType).toMatchSnapshot('light');

      const tmp = getTheme(props);
      expect(typeof tmp).toEqual('string');
      expect(tmp).toMatchSnapshot();
    });
  });
});
