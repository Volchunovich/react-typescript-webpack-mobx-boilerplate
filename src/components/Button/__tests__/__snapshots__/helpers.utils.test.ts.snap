// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Helpers Should received 'light' buttonType and return string 1`] = `
"
  // Light styles

  background: linear-gradient(to bottom,#a4d4ff 50%,#91c6f5 50%);
  border-bottom: 8px solid #4e0a71;
"
`;

exports[`Button Helpers Should received 'light' buttonType and return string: light 1`] = `"light"`;

exports[`Button Helpers Should received 'primary' buttonType and return string 1`] = `
"
  // Primary styles

  background: linear-gradient(to bottom,#9239d1 50%,#8239d1 50%);
  border-bottom: 8px solid #6b278e
"
`;

exports[`Button Helpers Should received 'primary' buttonType and return string: primary 1`] = `"primary"`;
